Sources/ComposableUserNotifications/LiveKey.swift

```diff
@@ -0,0 +1,126 @@
+import Foundation
+import Combine
+import ComposableArchitecture
+import UserNotifications
+
+extension UserNotificationClient: DependencyKey {
+  public static var liveValue: Self {
+    let center = UNUserNotificationCenter.current()
+
+    var client = UserNotificationClient()
+    client.add = { try await UNUserNotificationCenter.current().add($0) }
+
+    #if os(iOS) || os(macOS) || os(watchOS) || targetEnvironment(macCatalyst)
+    client.deliveredNotifications = {
+      let notifications = await center.deliveredNotifications()
+      return notifications.map(Notification.init(rawValue:))
+    }
+    #endif
+
+    client.notificationSettings = {
+      let settings = await center.notificationSettings()
+      return Notification.Settings(rawValue: settings)
+    }
+
+    #if os(iOS) || os(macOS) || os(watchOS) || targetEnvironment(macCatalyst)
+    client.notificationCategories = {
+      await center.notificationCategories()
+    }
+    #endif
+
+    client.pendingNotificationRequests = {
+      let requests = await center.pendingNotificationRequests()
+      return requests.map(Notification.Request.init(rawValue:))
+    }
+
+    #if os(iOS) || os(macOS) || os(watchOS) || targetEnvironment(macCatalyst)
+    client.removeAllDeliveredNotifications = {
+      center.removeAllDeliveredNotifications()
+    }
+    #endif
+
+    client.removeAllPendingNotificationRequests = {
+      center.removeAllPendingNotificationRequests()
+    }
+
+    #if os(iOS) || os(macOS) || os(watchOS) || targetEnvironment(macCatalyst)
+    client.removeDeliveredNotificationsWithIdentifiers = { identifiers in
+      center.removeDeliveredNotifications(withIdentifiers: identifiers)
+    }
+    #endif
+
+    client.removePendingNotificationRequestsWithIdentifiers = { identifiers in
+      center.removePendingNotificationRequests(withIdentifiers: identifiers)
+    }
+
+    client.requestAuthorization = { options in
+      try await center.requestAuthorization(options: options)
+    }
+
+    #if os(iOS) || os(macOS) || os(watchOS) || targetEnvironment(macCatalyst)
+    client.setNotificationCategories = { categories in
+      center.setNotificationCategories(categories)
+    }
+    #endif
+
+    client.supportsContentExtensions = {
+      center.supportsContentExtensions
+    }
+
+    client.delegate = {
+      AsyncStream { continuation in
+        let delegate = Delegate(continuation: continuation)
+        UNUserNotificationCenter.current().delegate = delegate
+        continuation.onTermination = { [delegate] _ in }
```

Comment:
Something like this will avoid a warning `LiveKey.swift:74:41 Capture 'delegate' was never used`
```swift
continuation.onTermination = { _ in
     _ = delegate
}
```

Reply:
Good point. Wonder why they choose to use the capture list in [isowords](https://github.com/pointfreeco/isowords/blob/2ae9ee5967559c8b7d21dfd9469db5f327c43a06/Sources/ComposableUserNotifications/LiveKey.swift#L12) and get the warning.